test/services/schema-file-updater.test.js

```diff
@@ -163,6 +164,85 @@ describe('services > schema-file-updater', () => {
     });
   });
 
+  it('should format action hooks', () => {
+    expect.assertions(6);
+
+    const INVALID_HOOKS = 'Action With Invalid Hooks';
+    const INVALID_LOAD_HOOK = 'Action With Invalid Load Hook';
+    const INVALID_CHANGE_HOOK = 'Action With Invalid Change Hook';
+    const VALID_HOOKS = 'Action With Valid Hooks';
+    const VALID_ONLY_LOAD_HOOK = 'Action With Valid Only a Load Hook';
+    const VALID_ONLY_CHANGE_HOOK = 'Action With Valid Only a Change Hook';
+
+    const schema = buildSchema([{
+      name: 'collectionName',
+      actions: [
+        {
+          name: INVALID_HOOKS,
+          type: 'single',
+          hooks: { load: 'oops', change: null },
+        },
+        {
+          name: INVALID_LOAD_HOOK,
+          type: 'single',
+          hooks: { load: 'oops', change: { foo: () => { } } },
+        },
+        {
+          name: INVALID_CHANGE_HOOK,
+          type: 'single',
+          hooks: { load: () => { }, change: null },
+        },
+        {
+          name: VALID_HOOKS,
+          type: 'single',
+          hooks: {
+            load: () => { },
+            change: {
+              foo: () => { },
+              bar: () => { },
+            },
+          },
+        },
+        {
+          name: VALID_ONLY_LOAD_HOOK,
+          type: 'single',
+          hooks: { load: () => { } },
+        },
+        {
+          name: VALID_ONLY_CHANGE_HOOK,
+          type: 'single',
+          hooks: {
+            change: {
+              foo: () => { },
+              bar: () => { },
+            },
+          },
+        },
+      ],
+    }]);
+
+    const { actions } = schema.collections[0];
+    const findAction = (name) => actions.find((action) => action.name === name);
+
+    expect(findAction(INVALID_HOOKS))
+      .toMatchObject({ hooks: { load: false, change: [] } });
+
+    expect(findAction(INVALID_LOAD_HOOK))
+      .toMatchObject({ hooks: { load: false, change: ['foo'] } });
+
+    expect(findAction(INVALID_CHANGE_HOOK))
+      .toMatchObject({ hooks: { load: true, change: [] } });
+
+    expect(findAction(VALID_HOOKS))
+      .toMatchObject({ hooks: { load: true, change: ['foo', 'bar'] } });
+
+    expect(findAction(VALID_ONLY_LOAD_HOOK))
+      .toMatchObject({ hooks: { load: true, change: [] } });
```

Comment:
shouldn't be `false` in this case?

Reply:
It seems the test is correct:

 - `VALID_ONLY_LOAD_HOOK ` is an _"Action With Only a Load Hook"_, so it aims to have a load hook but no change hook.
 - Then it is initialized with: `hooks: { load: () => { } },`
 - The result is what we expect:
   - `load` it `true` (because it has a function)
   - `change` is `[]` which is the default value of `change`

The objective was to have a `load` hook set and the default value for `change` hook, everything seems fine.

👉 I just improved the description of the actions